File: 0038-count-and-say/0038-count-and-say.js
/**
 * @param {number} n
 * @return {string}
 */
const countAndSay = (n) => {
  let str = '1';
  while (--n) {
    let res = '';
    let cnt = 0;
    let last = '';
    for (const d of str) {
      if (d !== last && cnt) {
        res = res + cnt + '' + last;
        cnt = 0;
      }
      ++cnt;
      last = d;
    }
    if (cnt) {
      res = res + cnt + '' + last;
    }
    str = res;
  }
  return str;
};

File: 0038-count-and-say/0038-count-and-say.go
import (
  "strconv"
)

func countAndSay(n int) string {
  ans := []rune{'1', 0}
  for range n-1 {
    arr := []rune{}
    cnt := 0
    for i, rn := range ans {
      if cnt > 0 && rn != ans[i-1] {
        arr = append(arr, append([]rune(strconv.Itoa(cnt)), ans[i-1])...)
        cnt = 0
      }
      cnt++
    }
    ans = append(arr, 0)
  }
  return string(ans[:len(ans)-1])
}